apps/remote_control/lib/lexical/remote_control/search/indexer/entry.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Entry do
+  @type entry_type :: :module | :function | :macro | :field
+  @type subject :: String.t()
+  @type entry_subtype :: :reference | :definition
+  @type indexer :: :beam | :source
+  @type line :: non_neg_integer()
+  @type column :: non_neg_integer
+  @type position :: {line, column}
+  @type version :: String.t()
+  @type entry_reference :: reference() | nil
+
+  defstruct [
```

Comment:
Something that may be missing from this is public/private. There are a _lot_ more `defp` than `def` in the average codebase, for instance. Within a module, we'd want results for private functions defined in that module. Looking up results from a remote module, though, we might not want to retrieve all private things too and have to filter them out after the fact.

Reply:
i'm not sure I follow; for find references, I'd like things to be exhaustive. I don't want to click on a module in some remote module and not seeing usages in private functions.

Reply:
That makes sense. I was thinking from a different perspective, like fuzzy completions instead of find references. So you might have something like `MyModule.thing|` and a possible completion might be `MyModule.get_thing` -- in that case, you're looking up definitions, not references, but it seems like it shouldn't/won't be hard to exclude private things after the fact.

Reply:
Correct, given how many times functions are called in a typical app, removing private functions will be better accomplished via a filtering step. 
I agree in general though, I'll add a `visibility` field that defaults to `public`.